pkg/kapp/diff/change_set_with_versioned_rs_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"testing"
+
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+)
+
+func TestChangeSet_ExistingVersioned_NewNonVersioned_Resource(t *testing.T) {
+	newRs, _ := ctlres.NewResourcesFromBytes([]byte(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret
+`))
+
+	existingRes, _ := ctlres.NewResourcesFromBytes([]byte(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-ver-1
+  annotations:
+    kapp.k14s.io/versioned: ""
+`))
+
+	changeSetWithVerRes := NewChangeSetWithVersionedRs(existingRes, newRs, nil, ChangeSetOpts{}, ChangeFactory{})
+
+	changes, err := changeSetWithVerRes.Calculate()
+
+	if err != nil {
+		t.Fatalf("Expected non-error")
+	}
+
+	if changes[0].Op() != ChangeOpDelete {
```

Comment:
before checking on changes[0] and changes[1], i would recommend to check on the length of changes (in this case should be 2). this would ensure that we are not getting some additional changes erroneously.